core/src/main/java/org/infinispan/notifications/impl/AbstractListenerImpl.java

```diff
@@ -357,61 +357,61 @@ public ListenerInvocationImpl(Object target, Method method, boolean sync, ClassL
          this.target = target;
          this.method = method;
          this.sync = sync;
-         this.classLoader = new WeakReference<ClassLoader>(classLoader);
+         this.classLoader = new WeakReference<>(classLoader);
          this.subject = subject;
       }
 
       @Override
       public void invoke(final A event) {
-         Runnable r = new Runnable() {
-
-            @Override
-            public void run() {
-               ClassLoader contextClassLoader = null;
-               Transaction transaction = suspendIfNeeded();
-               if (classLoader.get() != null) {
-                  contextClassLoader = SecurityActions.setContextClassLoader(classLoader.get());
-               }
-               try {
-                  if (subject != null) {
-                     try {
-                        Security.doAs(subject, new PrivilegedExceptionAction<Void>() {
-                           @Override
-                           public Void run() throws Exception {
-                              method.invoke(target, event);
-                              return null;
-                           }
-                        });
-                     } catch (PrivilegedActionException e) {
-                        Throwable cause = e.getCause();
-                        if (cause instanceof InvocationTargetException) {
-                           throw (InvocationTargetException)cause;
-                        } else if (cause instanceof IllegalAccessException) {
-                           throw (IllegalAccessException)cause;
-                        } else {
-                           throw new InvocationTargetException(cause);
+         Runnable r = () -> {
+            ClassLoader contextClassLoader = null;
+            Transaction transaction = suspendIfNeeded();
+            if (classLoader.get() != null) {
+               contextClassLoader = SecurityActions.setContextClassLoader(classLoader.get());
+            }
+            try {
+               if (subject != null) {
+                  try {
+                     Security.doAs(subject, (PrivilegedExceptionAction<Void>) () -> {
+                        if (trace) {
+                           // Don't want to print out Subject as it could have sensitive information
+                           log.tracef("Invoking listener: %s passing event %s using subject", target, event);
```

Comment:
I find it pretty annoying when I search for a log message in a class and I can't find it because it was actually logged by a base class, so I'm not a big fan of the abstract `getLog()` strategy, but I guess if it's already there it's best to use the same approach everywhere.